ykcapi/src/lib.rs

```diff
@@ -136,13 +136,23 @@ pub struct CVec {
     length: usize,
 }
 
+/// Address, offset, and length of the live AOT values for this guard failure. Mirrors the struct
+/// defined in ykllvmwrap/jitmodbuilder.cc.
+#[derive(Debug)]
+#[repr(C)]
+pub struct LiveAOTVals {
+    addr: *const c_void,
+    offset: usize,
+    length: usize,
+}
+
 /// Reads the stackmap and saved registers from the given address (i.e. the return address of the
 /// deoptimisation call).
 #[cfg(target_arch = "x86_64")]
 #[no_mangle]
 pub extern "C" fn yk_stopgap(
     stackmap: &CVec,
-    aotmap: &CVec,
+    aotvals: &LiveAOTVals,
```

Comment:
Small nit: would it be better to receive this argument as a raw pointer? We are only going to cast it in a moment anyway.

Reply:
I don't follow. It's a struct and I want to read out its fields. Passing this as a raw pointer would make reading out those fields much less nice.

Reply:
Sorry, I misread.